src/box/txn_limbo.c

```diff
@@ -918,7 +924,8 @@ txn_limbo_on_parameters_change(struct txn_limbo *limbo)
 			assert(confirm_lsn > 0);
 		}
 	}
-	if (confirm_lsn > limbo->confirmed_lsn && !limbo->is_in_rollback) {
+	if (confirm_lsn > limbo->confirmed_lsn &&
+	    !limbo->is_in_rollback && !limbo->frozen) {
```

Comment:
Can you move `limbo->frozen` check before the foreach loop?
Seems you could do nothing at all (not even fiber_cond_broadcast) when limbo is frozen.

Reply:
Fixed